coord/coordinator.go

```diff
@@ -229,27 +336,22 @@ func (c *Coordinator[K, A]) attemptSendMessage(ctx context.Context, protoID addr
 func (c *Coordinator[K, A]) StartQuery(ctx context.Context, queryID query.QueryID, protocolID address.ProtocolID, msg kad.Request[K, A]) error {
 	knownClosestPeers := c.rt.NearestNodes(msg.Target(), 20)
 
-	ev := &eventAddQuery[K, A]{
+	qev := &query.EventPoolAddQuery[K, A]{
 		QueryID:           queryID,
 		Target:            msg.Target(),
 		ProtocolID:        protocolID,
 		Message:           msg,
-		KnownClosestPeers: knownClosestPeers,
+		KnownClosestNodes: knownClosestPeers,
 	}
-
-	c.queue.Enqueue(ctx, ev)
-	// c.inboundEvents <- ev
-
+	c.pool.Enqueue(ctx, qev)
```

Comment:
What if the queue is at its capacity? `Enqueue` is exported so anyone could put messages in the queue. If it was just internal we probably could make sure that this won't happen.

Besides in the coordinator, `Enqueue` is only called in tests, so we could make it a private method?

Reply:
> What if the queue is at its capacity? Enqueue is exported so anyone could put messages in the queue. If it was just internal we probably could make sure that this won't happen.

The pool is only internal - the coordinator creates it - so noone else is adding events. 

> Besides in the coordinator, Enqueue is only called in tests, so we could make it a private method?

Do you mean pool.Enqeue? That's from synchronous methods exposed by the coordinator, not just tests.